core/src/main/scala/io/snappydata/SnappyDaemons.scala

```diff
@@ -111,22 +112,27 @@ object SnappyAnalyticsService extends Logging {
     tablename.startsWith(INTERNAL_SCHEMA_NAME) && tablename.endsWith(SHADOW_TABLE_SUFFIX)
 
 
-  def getTableSize(tableName: String, isColumnTable: Boolean = false): Long = {
+  def getTableSize(sqlContext: SQLContext, tableName: String, isColumnTable: Boolean = false):
+  Long = {
     val currentTableStats = tableStats
     if (currentTableStats == null || !currentTableStats.contains(tableName)) {
-      defaultStats.valueSize
+      defaultSizeForPlanning(sqlContext)
     }
     else {
       if (isColumnTable) {
         currentTableStats.get(ColumnFormatRelation.cachedBatchTableName(tableName))
-            .getOrElse(defaultStats).valueSize
+            .getOrElse(defaultSizeForPlanning(sqlContext))
         +currentTableStats.get(tableName).get.valueSize
       } else {
         currentTableStats.get(tableName).get.valueSize
       }
     }
   }
 
+  def defaultSizeForPlanning(sqlContext: SQLContext) : Long =
+    (sqlContext.getConf("spark.sql.autoBroadcastJoinThreshold") + 1).toLong
+
```

Comment:
Better to use: sqlContext.conf.defaultSizeInBytes

Also I think it will be cleaner to not do this here at all and instead let getTableSize return Option[Long], and in the calling points:

```scala
  override def sizeInBytes: Long = SnappyAnalyticsService.getTableSize(table,
    isColumnTable = true).getOrElse(super.sizeInBytes)
```

What do you think?